backend/de.metas.business/src/main/sql/postgresql/ddl/functions/Reset_System_Base_Language.sql

```diff
@@ -0,0 +1,78 @@
+CREATE OR REPLACE FUNCTION Reset_System_Base_Language(p_source_language      varchar,
+                                                      p_destination_language varchar) RETURNS VOID
+AS
+$$
+DECLARE
+    destination_language  varchar := '';
+    source_language       varchar := '';
+    table_name            varchar;
+    base_table_id         numeric;
+    base_table            varchar;
+    query_set_template    text    := '';
+    final_update_query    text    := '';
+    v_columnname          varchar;
+    update_query_template varchar := 'WITH X AS (SELECT * FROM base_table_trl WHERE base_table_id = %s AND ad_language =  ''%s'') '
+                                         || 'UPDATE base_table as target '
+                                         || 'SET %s '
+                                         || 'FROM X '
+        || 'WHERE target.base_table_id = x.base_table_id ;' ;
+BEGIN
+
+    IF (p_source_language IS NULL OR TRIM(p_source_language) = '') THEN
+        RAISE WARNING 'No source language is provided';
+        RETURN;
+    END IF;
+
+    IF (p_destination_language IS NULL OR TRIM(p_destination_language) = '') THEN
+        RAISE WARNING 'No destination language is provided';
+        RETURN;
+    END IF;
+
+    source_language := TRIM(p_source_language);
+    destination_language := TRIM(p_destination_language);
+
+    -- update languages
+    UPDATE ad_language
+    SET isbaselanguage='N', issystemlanguage='Y'
+    WHERE ad_language = source_language;
+
+    UPDATE ad_language
+    SET isbaselanguage = 'Y', issystemlanguage = 'Y'
+    WHERE ad_language = destination_language;
+
+    -- Iterate over Trls tables
+    FOR table_name IN SELECT UPPER(tablename) FROM ad_table WHERE UPPER(tablename) LIKE UPPER('%_Trl') ORDER BY 1
+        LOOP
+            base_table := REPLACE(table_name, '_TRL', '');
+            BEGIN
+                -- Reset query output
+                query_set_template := '';
+                FOR v_columnname IN SELECT columnname FROM ad_column WHERE istranslated = 'Y' AND ad_table_id = get_table_id(base_table)
```

Comment:
i'm not really sure about this part
when teo mentioned `IsTranslated` yesterday, i assumed that it was about e.g. `M_Product_Trl.IsTranslated`. 
Meaning, AFAIU Teo's advise was to only copy e.g. `M_Product_Trl.Name` => `M_Product.Name `if the respective `M_Product_Trl` record had `IsTranslated='Y'`.

Regarding `AD_Column.IsTranslated` - i'm not sure what it is about. You seem to assume that all `AD_Columns` which have a pendant in a `_Trl` table have this set to `Y`...IDK if that'S the case..
If not, you would need to check the columns of the Trl table against those of the main table copy the string/text/etc columns..

Reply:
@Chemssou  cc @metas-ts 

agree with @Chemssou about how he handles AD_Column.IsTranslated.
That's how metasfresh knows which columns have translations.
See https://github.com/metasfresh/metasfresh/blob/187d7e339aa8c1f70d0b11739160775a73511bce/backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/POInfo.java#L306-L305


About 
> Meaning, AFAIU Teo's advise was to only copy e.g. M_Product_Trl.Name => M_Product.Name if the respective M_Product_Trl record had IsTranslated='Y'.

No, i didn't. I've advised about AD_Column.IsTranslated.
About this idea, sounds good but it's not feasible because in the last years we didn't pay attention to `*_Trl.IsTranslated` flags.
So they are kind of not maintained...